### PR TITLE
feat: Oauth 2.0 with Google

### PR DESCRIPTION
**Pull request by M-Byte480**
There is a `./frontend` in the root:
image

This is just a hyperlink to Google Oauth for our application:
image

THIS WILL NOT WORK FOR YOU UNLESS I ADD YOU AS A TEST USER TO GOOGLE SYSTEM:
image
Upon successful authentication we get the pair of Access and Refresh token. We note the expirations in the DB and at the moment there is a get request with the user details:
image
If the user does not exists, we note them in the DB as a new user.

There are two secrets that I need to share with you to run the application: Client_id and client_secret

Will write more unit tests for the Google Authentication Service, as much as I disagree with testing APIs, the coverage is only at 50% with not testing it -_-

### CODE REVIEW

**Review by silpac-12 — APPROVED** [review]

File: Microservices/TemplateProject/src/main/java/cs4337/group_8/AuthenticationMicroservice/DTO/SampleDTO.java
Comment (Silver-Golden) [inline Microservices/TemplateProject/src/main/java/cs4337/group_8/AuthenticationMicroservice/DTO/SampleDTO.java:1]: I think ye may have messed up something here considering that this is being moved not copied.

File: frontend/index.html
Comment (Silver-Golden) [inline frontend/index.html:1]: Looks good

**Review by Silver-Golden — CHANGES_REQUESTED** [review]
Looks good, except for teh moving of files.

You may also want to do this to fetch the latest changes from main.

```bash
git fetch origin main
git rebase origin/main
```

File: frontend/index.html
Comment (M-Byte480) [reply frontend/index.html:1]: thanks

File: Microservices/TemplateProject/src/main/java/cs4337/group_8/AuthenticationMicroservice/DTO/SampleDTO.java
Comment (M-Byte480) [reply Microservices/TemplateProject/src/main/java/cs4337/group_8/AuthenticationMicroservice/DTO/SampleDTO.java:1]: I believe this file no longer exists.

**Review by Silver-Golden — APPROVED** [review]
Looks good to me!

**Pull request merged**